Mathlib/Analysis/SpecialFunctions/Pow/Complex.lean

```diff
@@ -112,53 +112,40 @@ theorem cpow_sub {x : ℂ} (y z : ℂ) (hx : x ≠ 0) : x ^ (y - z) = x ^ y / x
 theorem cpow_neg_one (x : ℂ) : x ^ (-1 : ℂ) = x⁻¹ := by simpa using cpow_neg x 1
 #align complex.cpow_neg_one Complex.cpow_neg_one
 
+lemma cpow_int_mul (x : ℂ) (n : ℤ) (y : ℂ) : x ^ (n * y) = (x ^ y) ^ n := by
+  rcases eq_or_ne x 0 with rfl | hx
+  · rcases eq_or_ne n 0 with rfl | hn
+    · simp
+    · rcases eq_or_ne y 0 with rfl | hy <;> simp [*, zero_zpow]
+  · rw [cpow_def_of_ne_zero hx, cpow_def_of_ne_zero hx, mul_left_comm, exp_int_mul]
+
+lemma cpow_mul_int (x y : ℂ) (n : ℤ) : x ^ (y * n) = (x ^ y) ^ n := by rw [mul_comm, cpow_int_mul]
+
+lemma cpow_nat_mul (x : ℂ) (n : ℕ) (y : ℂ) : x ^ (n * y) = (x ^ y) ^ n := by
+  exact_mod_cast cpow_int_mul x n y
+
+lemma cpow_mul_nat (x y : ℂ) (n : ℕ) : x ^ (y * n) = (x ^ y) ^ n := by
+  rw [mul_comm, cpow_nat_mul]
+
 @[simp, norm_cast]
-theorem cpow_nat_cast (x : ℂ) : ∀ n : ℕ, x ^ (n : ℂ) = x ^ n
-  | 0 => by simp
-  | n + 1 =>
-    if hx : x = 0 then by
-      simp only [hx, pow_succ, Complex.zero_cpow (Nat.cast_ne_zero.2 (Nat.succ_ne_zero _)),
-        zero_mul]
-    else by simp [cpow_add, hx, pow_add, cpow_nat_cast x n]
+theorem cpow_nat_cast (x : ℂ) (n : ℕ) : x ^ (n : ℂ) = x ^ n := by simpa using cpow_nat_mul x n 1
 #align complex.cpow_nat_cast Complex.cpow_nat_cast
 
 @[simp]
-theorem cpow_two (x : ℂ) : x ^ (2 : ℂ) = x ^ (2 : ℕ) := by
-  rw [← cpow_nat_cast]
-  simp only [Nat.cast_ofNat]
+lemma cpow_ofNat (x : ℂ) (n : ℕ) [n.AtLeastTwo] :
```

Comment:
Isn’t this true for 0 and 1 as well?

Reply:
We have the following `OfNat` instances in `Mathlib`:

- `[One R] : OfNat R 1 := ⟨One.one⟩` [here](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Init/ZeroOne.html#One.toOfNat1)
- `[Zero R] : OfNat R 0 := ⟨Zero.zero⟩` [here](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Init/ZeroOne.html#Zero.toOfNat0)
- `[NatCast R] [n.AtLeastTwo] : OfNat R n := ⟨Nat.cast n⟩`

so lemmas about `ofNat n` don't typecheck without `[n.AtLeastTwo]`.

Reply:
I’m quite excited to use this type class now. :)

https://github.com/girving/ray/blob/1c46b8324e991d6d82bb50fdf93c821599c94314/Ray/Dynamics/Mandelbrot.lean#L24